### PR TITLE
QPID-8161:[Broker-J]wrap locktimeout as connectionScoped Exception

### PR DESCRIPTION
**Pull request by Dedeepya-T**
Wrapped the lock timeout exception as a Connection scoped exception

### CODE REVIEW

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/BDBMessageStoreTest.java [hunk 1]
```diff
@@ -218,4 +228,25 @@ protected boolean flowToDiskSupported()
         return true;
     }
 
+    @Test
+    public void testGetNextMessgaeIdWithLockException(){
+        BDBMessageStore bdbStore = (BDBMessageStore) createMessageStore();
```
Comment (alex-rufous) [inline bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/BDBMessageStoreTest.java:233]: The test mixes different mocking approaches: spy, mockito moc and mocked messages for AMQP 0-8.
IMHO, we can get rid of spy by using mockito mocks for EnvironmentFacadeFactory, EnvironmentFacade and message
```
EnvironmentFacade eff = mock(EnvironmentFacadeFactory.class);
EnvironmentFacade ef = mock(EnvironmentFacade.class);
doThrow(LockTimeoutException.class).when(ef).openSequence(any(),any(),any());
when(eff.createEnvironmentFacade(any())).thenReturn(ef);

BDBMessageStore store = new BDBMessageStore (eff);
store.openMessageStore();

 try
        {
            store.addMessage(new InternalMessageMetaData(false, null, 0))
            fails("exception is expected");
        }
        catch(ConnectionScopedRuntimeException e)
        {
            assertEquals("Unexpected exception on BDB sequence",e.getMessage());
        }

```

I think the above reduces dependency to amqp 0-8 protocol and gets rid of spy

**Review by alex-rufous — CHANGES_REQUESTED** [review]

File: bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/BDBMessageStoreTest.java (hunk 1 above)
Comment (Dedeepya-T) [reply bdbstore/src/test/java/org/apache/qpid/server/store/berkeleydb/BDBMessageStoreTest.java:233]: Changed it!

File: bdbstore/src/main/java/org/apache/qpid/server/store/berkeleydb/AbstractBDBMessageStore.java [hunk 2]
```diff
@@ -231,6 +233,10 @@ public long getNextMessageId()
                                                               MESSAGE_METADATA_SEQ_CONFIG);
             newMessageId = mmdSeq.get(null, 1);
         }
+        catch(LockTimeoutException le)
+        {
+           throw new ConnectionScopedRuntimeException("Unexpected exception on BDB sequence",le);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply bdbstore/src/main/java/org/apache/qpid/server/store/berkeleydb/AbstractBDBMessageStore.java:238]: Done!

**Review by alex-rufous — APPROVED** [review]
the changes look good to me. I'll commit them shortly

**Pull request closed without merging**